File: apps/app/lib/api/types/agent.ts
export interface ApiAgentTemplate {
  id: string;
  name: string;
  slug: string;
  description: string;
  category: string;
  systemPrompt: string;
  suggestedModel: string;
  temperature: number;
  maxTokens: number;
  builtinToolNames?: string[];
  mcpServerSlugs?: string[];
  capabilities?: string[];
  tags?: string[];
  author: string;
  isOfficial: boolean;
  createdAt: string;
}

export interface ApiAgent {
  id: string;
  name: string;
  displayName?: string;
  slug: string;
  description: string;
  systemPrompt: string;
  systemPromptOverride?: string;
  model: string;
  agentType?: string;
  temperature: number;
  maxTokens: number;
  maxBudgetCents?: number;
  allowedModels?: string[];
  allowedTools?: string[];
  allowedResources?: string[];
  allowedMcpServers?: string[];
  toolNames?: string[];
  mcpServerIds?: string[];
  routingPolicyId?: string;
  enabled: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface ApiCreateAgentRequest {
  name: string;
  displayName?: string;
  slug?: string;
  description?: string;
  systemPrompt?: string;
  model?: string;
  agentType?: string;
  temperature?: number;
  maxTokens?: number;
  maxBudgetCents?: number;
  allowedModels?: string[];
  allowedTools?: string[];
  allowedResources?: string[];
  allowedMcpServers?: string[];
  toolNames?: string[];
  mcpServerIds?: string[];
  routingPolicyId?: string;
  enabled?: boolean;
}

export interface ApiAgentStats {
  totalRequests: number;
  totalTokens: number;
  totalCostUSD: number;
  avgLatencyMs: number;
  successRate: number;
  toolCallsCount: number;
}

File: apps/api/internal/models/agent.go
package models

import (
	"time"
)

type Agent struct {
	ID                   string    `json:"id"`
	UserID               string    `json:"userId"`
	Name                 string    `json:"name"`
	DisplayName          string    `json:"displayName"`
	Description          string    `json:"description"`
	AgentType            string    `json:"agentType"`
	SystemPromptOverride string    `json:"systemPromptOverride"`
	AllowedModels        []string  `json:"allowedModels"`
	AllowedTools         []string  `json:"allowedTools"`
	AllowedResources     []string  `json:"allowedResources"`
	AllowedMCPServers    []string  `json:"allowedMcpServers"`
	MaxBudgetCents       int       `json:"maxBudgetCents"`
	Status               string    `json:"status"`
	Enabled              bool      `json:"enabled"`
	CreatedAt            time.Time `json:"createdAt"`
	UpdatedAt            time.Time `json:"updatedAt"`
}

type AgentGovernanceCheckResult struct {
	AgentName       string `json:"agentName"`
	ModelAllowed    bool   `json:"modelAllowed"`
	ToolAllowed     bool   `json:"toolAllowed"`
	ResourceAllowed bool   `json:"resourceAllowed"`
	Reason          string `json:"reason,omitempty"`
}

type AgentStats struct {
	TotalRequests  int     `json:"totalRequests"`
	TotalTokens    int     `json:"totalTokens"`
	TotalCostUSD   float64 `json:"totalCostUSD"`
	AvgLatencyMs   int     `json:"avgLatencyMs"`
	SuccessRate    float64 `json:"successRate"`
	ToolCallsCount int     `json:"toolCallsCount"`
}
